### PR TITLE
Cosmos Spark: Fix dependency issues in Synapse

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description
Fixes dependency issues in Synapse by correcting shading instructions.
 
# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #### Bugs Fixed
 
 #### Other Changes
+* Fixed shading instructions to correct dependency issues in Azure Synapse with version 4.14.0 and 4.14.1. - See [PR 31980](https://github.com/Azure/azure-sdk-for-java/pull/31980)
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md:12]: does this issue only happen > 4.14.0? is it a version upgrade result?

should the changelog also be added in 3-3?

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md:12]: I don't think this change is relevant for Spark 3-3 - we never published Spark 3-3 to Maven with broken shading instructions?

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md:12]: 4.13.1 still worked fine (used that with the internal customer hitting the dependency errors when I asked them to upgrade Spark connector to latest version)

Comment (xinlian12) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md:12]: gotcha, thanks for the context :)

**Pull request merged**